src/dos/program_more.cpp

```diff
@@ -47,46 +48,72 @@ void MORE::Run()
 	}
 
 	MoreOutputFiles output(*this);
-	if (!ParseCommandLine(output) || shutdown_requested) {
+	if (!ParseCommandLine(output) || !FindInputFiles(output) ||
+	    shutdown_requested) {
 		return;
 	}
 	output.Display();
 }
 
 bool MORE::ParseCommandLine(MoreOutputFiles &output)
 {
-	// Put all the parameters into vector
-	std::vector<std::string> params;
-	cmd->FillVector(params);
+	if (cmd->FindExist("/c", true)) {
+		output.SetOptionClear(true);	
+	}
+	if (cmd->FindExist("/e", true)) {
+		output.SetOptionExtendedMode(true);	
+	}
+	if (cmd->FindExist("/p", true)) {
+		output.SetOptionExpandFormFeed(true);
+	}
+	if (cmd->FindExist("/s", true)) {
+		output.SetOptionSquish(true);
```

Comment:
This is a nice improvement; thanks for digging into the CommandArgs route.

One small suggestion: in `cmd->FindExist("/c", true)`, the `true` has always been non-self-documenting.

Suggest hoisting it up to a named `constexpr bool` that indicates the argument will be eaten/consumed/removed if found.

Reply:
For the `output.Set(bool)` calls, if a user doesn't provide the argument, then does it switch them off/false?

Would something like this be more direct and more guaranteed? 

```c++
auto cli_has() = [&](const char* arg) {
    constexpr bool remove_if_found = true;
    return cmd->FindExist(arg, remove_if_found));
};
output.SetOptionClear(cli_has("/c"));
output.SetOptionExtendedMode(cli_has("/e"));
output.SetOptionExpandFormFeed(cli_has"/p"));
output.SetOptionSquish(cli_has("/s"));
```

This avoids assumptions about the incoming prior MoreOutputFiles Clean/ExtendedMode/ExpandedForm/Squish option states.

Reply:
Ok, changed.